core/commands/partials/terraform_configurator.rb

```diff
@@ -137,12 +137,16 @@ def bring_up_and_configure(node)
   def store_network_settings(node)
     @ui.info('Generating network configuration file')
     begin
+      network_result = TerraformService.resource_network(node, @ui, @config.path)
+      raise(network_result.error) if network_result.error?
```

Comment:
You should instead of throwing and caching the exception use `and_then` method of the Result.